### ISSUE TITLE
ExplicitItLambdaParameter false positive for un-inferrable lambda

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#6364]
## Expected Behavior
No report when removing would result in compilation error.

## Observed Behavior
The below code in repro flags ``Detekt - ExplicitItLambdaParameter: This explicit usage of `it` as the lambda parameter name can be omitted.``, and if I do omit I get:
![image](https://github.com/detekt/detekt/assets/2906988/4bebc2d1-3299-4e29-aeeb-2ed786b463b3)
because the inference from the return type doesn't pass through `::invoke`, maybe in K2? but more likely not.

## Steps to Reproduce
```kotlin
operator fun <T> KFunction1<T, Boolean>.not(): KFunction1<T, Boolean> =
	{ it: T -> !this(it) }::invoke
```

## Context
https://stackoverflow.com/a/76828982/253468, detekt flags all lambdas before `::invoke`.

## Your Environment

* Version of detekt used:
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by atulgpt** [comment]
Hi @TWiStErRob more simpler example would be as follow
```
fun test(): (Int) -> Int {
    return { it: Int -> it.inc() }::invoke
}
```
